### DISCUSSION TITLE
Upgrade project from php 7.3 to PHP 7.4. afterwords upgrade to PHP 8.0.

### DISCUSSION THREAD

**Discussion by Morpheus-ro** [rectorphp/rector#8749] [Q&A] [answered]
Hello everybody,

I am new into using Rector and I want to get my project upgraded to the latest PHP version. Current version is PHP 7.3.
Currently I have this rector.php config file and I just want to be sure that this is correct:

`
return RectorConfig::configure()
    ->withPaths([
        __DIR__ . '/application',
        __DIR__ . '/model'
    ])
    // uncomment to reach your current PHP version
    ->withPhpSets(
        false,
        false,
        false,
        false,
        false,
        true
    )
    ->withTypeCoverageLevel(0);
`

my goal here is to get to PHP 7.4 and then to PHP 8.0 and so on.
Any ideas, suggestions over the config and over all?

**Comment by samsonasik** [comment, marked as answer]
If you still on php 7.x, you can use `LevelSetList` instead:

```php
use Rector\Config\RectorConfig;
use Rector\Set\ValueObject\LevelSetList;

return RectorConfig::configure() 
    ->withPaths([ __DIR__ . '/application', __DIR__ . '/model' ]) 
    ->withSets([
          LevelSetList::UP_TO_PHP_53
     ]);
```

start with lower php 5.3 to see the change per level :)

**Comment by Morpheus-ro** [comment]
in my case, another colleague did this several years ago so we are now at php 7.3 so we just need to upgrade the version to php 8.3 at the end.

Thank you for you reply and have a nice day :)
